### PR TITLE
feat: activity ticker

### LINKED ISSUE #1457: Activity feed in Homepage v1
This feature introduces the concept of the **Activity Feed** on the DAO Homepage. The **Activity Feed** will serve as a continuous stream of the activity members of the DAO execute in the context of our Governance dApp. This feature will be built and released in several slices improving the capabilities offered to DAO members.

# V1 Release
- A new table for Events will be created using this structure:

## Events (new table)
- address
- username
- event type (voted | proposal_created | update_created)
- created_at
- event data

### update
- update id
- proposal id
- proposal title

### vote
- proposal id
- proposal title
- choice (TBD with gino)

### proposal
- proposal title
- proposal id

The activity feed will display the events declared on this document: https://docs.google.com/spreadsheets/d/1tfZsv62UYK2wzjtostIV6SCOJgeUyP8DZ-mYrH0NkWk/edit#gid=0

A new endpoint to get events will be created. The endpoint will return the last 7 days (168h)  of events.
New events endpoint
Returns last 7 days' events

## Frontend
Design: 
Desktop only? Mobile TBD.

### PR DESCRIPTION
**Pull request by 1emu**
**Screenshots**
image

image

**Related Issue**
Closes #1457

- [ ]  **Functionality validation**
    - [ ]  Feature working as described in the related issue
    - [ ]  New environment variables have been set
    - [ ]  New API services methods have tests
    - [ ]  Code errors are being logged in the code health service
- [ ]  **Product validation**
    - [ ]  UI changes look the same as in Figma
    - [ ]  Usage events are being sent to the product analytics service
    - [ ]  New common components have been documented in Storybook
    - [ ]  Preview link if it’s a customer facing issue
- [ ] **Documentation**
    - [ ] README updated
    - [ ] New API changes are documented
- [ ] **Accessibility**
    - [ ] UI changes are accessible

### CODE REVIEW

File: src/back/services/update.ts [hunk 1]
```diff
@@ -55,4 +62,106 @@ export class UpdateService {
...
+        updated_at: now,
+      },
+      { id }
+    )
+
+    const updatedUpdate = await UpdateService.getById(id)
+    if (updatedUpdate) {
+      if (!completion_date) {
+        await DiscourseService.createUpdate(updatedUpdate, proposal.title)
+        await EventsService.updateCreated(update.id, proposal.id, proposal.title, user)
```
[the remark this answers is not in the document]
Comment (andyesp) [reply src/back/services/update.ts:158]: It is ok but it's not necessarily vesting-related.

File: src/hooks/useProfiles.ts [hunk 2]
```diff
@@ -1,39 +1,23 @@
...
+import { DclProfile } from '../utils/Catalyst/types'
 
 import { DEFAULT_QUERY_STALE_TIME } from './constants'
 
-type Profile = {
-  profile: Avatar
-  isDefaultProfile: boolean
-}
-
 export default function useProfiles(addresses: (string | null | undefined)[]): {
```
Comment (andyesp) [inline src/hooks/useProfiles.ts:9]: Maybe `useDclProfiles`, `useMultipleDclProfiles`?

File: src/back/routes/events.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+async function getLatestEvents() {
+  return await EventsService.getLatest()
+}
+
+async function voted(req: WithAuth) {
+  const user = req.auth!
+
+  validateProposalId(req.body.proposalId)
+  validateRequiredString('proposalTitle', req.body.proposalTitle)
+  validateRequiredString('choice', req.body.choice)
```
Comment (ncomerci) [inline src/back/routes/events.ts:23]: I wonder if we shouldn't use the validation schema we use for proposals here as well?

File: src/back/services/events.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      acc[profile.address] = profile
+      return acc
+    }, {} as Record<string, DclProfile>)
+
+    const latestEventsWithAuthor: EventWithAuthor[] = []
+    for (const event of latestEvents) {
+      const { address } = event
+      latestEventsWithAuthor.push({
+        author: addressesToProfile[address].username || address,
+        avatar: addressesToProfile[address].avatar,
```
Comment (ncomerci) [inline src/back/services/events.ts:27]: We should return something even if `getProfiles` fails for some reason (using the address and a default avatar)

```
author: addressesToProfile[address]?.username || address,
avatar: addressesToProfile[address]?.avatar || DEFAULT_AVATAR,
```

File: src/back/services/events.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import { ErrorService } from '../../services/ErrorService'
+import { EventWithAuthor } from '../../shared/types/events'
+import { EventType, ProposalCreatedEvent, UpdateCreatedEvent, VotedEvent } from '../../shared/types/events'
+import { getProfiles } from '../../utils/Catalyst'
+import { DclProfile } from '../../utils/Catalyst/types'
+import { ErrorCategory } from '../../utils/errorCategories'
+import EventModel from '../models/Event'
+
+export class EventsService {
+  static async getLatest(): Promise<EventWithAuthor[]> {
```
Comment (ncomerci) [inline src/back/services/events.ts:12]: This fn could fail and has no trycatch

File: src/back/services/events.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+  private static reportEventError(error: Error, eventType: EventType, args: Record<string, unknown>) {
+    ErrorService.report('Error creating event', {
+      error,
+      event_type: eventType,
+      ...args,
+      category: ErrorCategory.Events,
+    })
+  }
+}
```
Comment (ncomerci) [inline src/back/services/events.ts:93]: Can we agree to use this pattern to report errors in services?

```typescript
if (isProdEnv()) {
   ErrorService.report(...)
} else {
   console.error(error)
}
```

File: src/back/routes/events.ts (hunk 3 above)
Comment (andyesp) [reply src/back/routes/events.ts:23]: Let's use zod for validation non-proposals schemas after merging the financial report pr.

**Review by ncomerci — APPROVED** [review]

File: src/utils/Catalyst/index.ts [hunk 7]
```diff
@@ -3,22 +3,31 @@ import isEthereumAddress from 'validator/lib/isEthereumAddress'
...
 
   const response: ProfileResponse = await (await fetch(`${CATALYST_URL}/lambdas/profile/${address}`)).json()
-
-  return response.avatars.length > 0 ? response.avatars[0] : null
+  const profile = response.avatars.length > 0 ? response.avatars[0] : null
+  return getDclProfile(profile, address)
 }
 
-export async function getProfiles(addresses: string[]): Promise<(Avatar | null)[]> {
+// TODO: this could be cached per address
```
Comment (andyesp) [inline src/utils/Catalyst/index.ts:29]: I'd remove this comment as this is the Catalyst client API. We should let the cache decision to whoever want to use it.

**Review by andyesp — APPROVED** [review]

**Pull request merged**